### PR TITLE
only run uninstall functions if dependencies are available

### LINKED ISSUE #2657: Can't delete plugin if WooCommerce is disabled
We got several reports of users not being able to delete the plugin in the forums:

https://wordpress.org/support/topic/cant-delete-it-3/

https://wordpress.org/support/topic/very-bad-cant-delete/

It looks like the issue only happens when WooCommerce is inactive (thanks @panoslyrakis for finding this!).

**To Reproduce**

1. Install WooCommerce and enable it.
2. Install WooCommerce Admin and enable it.
3. Disable WooCommerce.
4. Disable WooCommerce Admin.
5. Delete WooCommerce Admin.

Notice the process never ends.

**Expected behavior**
The plugin should be removed.

### PR DESCRIPTION
**Pull request by rrennick**
Fixes #2657

This PR adds a dependency check to the uninstall script. If WooCommerce is not active, the plugin is uninstalled without clearing the plugin specific data. It is unlikely that a site owner with significant WC data will be removing this plugin while WC is inactive. For the folks test driving WC & WCA this will not leave a significant amount of data behind.

### Detailed test instructions:

- Build the production version of the plugin
- Install it in a test install
- Activate both WC & WCA
- Deactivate WC (WCA should also deactivate)
- Delete WCA through to plugin screen

### Changelog Note:

Fix: Allow WooCommerce Admin to be deleted through the plugin screen when WooCommerce is not active.

### CODE REVIEW

File: uninstall.php [hunk 1]
```diff
@@ -9,5 +9,7 @@
 
 require_once dirname( __FILE__ ) . '/woocommerce-admin.php';
 
-WC_Admin_Feature_Plugin::instance()->includes();
-WC_Admin_Install::delete_table_data();
+if ( WC_Admin_Feature_Plugin::instance()->check_dependencies() ) {
```
Comment (jeffstieler) [inline uninstall.php]: It looks like `WC_Admin_Install::delete_table_data()` has no dependencies - perhaps we can just include its source file and call it?

Comment (rrennick) [reply uninstall.php]: It does have dependencies:

```
PHP Fatal error:  Uncaught Error: Class 'WC_Admin_Install' not found in /wp-content/plugins/woocommerce-admin/uninstall.php:15
```

Comment (rrennick) [reply uninstall.php]: > It does have dependencies:

Ignore that. I was relying on the autoloader. Addressed in https://github.com/woocommerce/woocommerce-admin/pull/2762/commits/8676faee9e2aeecdd1c054fff31d67af33a08c17

**Comment by rrennick** [comment]
@jeffstieler I removed the dependency check and restored the visibility of `check_dependencies()`.

**Pull request merged**